blocks/math.js

```diff
@@ -82,12 +82,19 @@ Blockly.Blocks['math_positive_number'] = {
    * @this Blockly.Block
    */
   init: function() {
-    this.setHelpUrl(Blockly.Msg.MATH_NUMBER_HELPURL);
-    this.setColour(Blockly.Blocks.math.HUE);
-    this.appendDummyInput()
-        .appendField(new Blockly.FieldNumber('0', 0, Infinity), 'NUM');
-    this.setOutput(true, 'Number');
-    this.setOutputShape(Blockly.OUTPUT_SHAPE_ROUND);
+    this.jsonInit({
+      "message0": "%1",
+      "args0": [
+        {
+          "type": "field_number",
+          "name": "NUM",
+          "min": 0
```

Comment:
Here's a relevant snippet [from](https://github.com/LLK/scratch-flash/blob/master/src/scratch/ScratchSprite.as#L274-L294) scratch-flash:

```as3
public function setPenHue(n:Number):void {
  penHue = n % 200;
  if (penHue < 0) penHue += 200;
  updateCachedPenColor();
}

public function setPenShade(n:Number):void {
  penShade = n % 200;
  if (penShade < 0) penShade += 200;
  updateCachedPenColor();
}

private function updateCachedPenColor():void {
  var c:int = Color.fromHSV((penHue * 180) / 100, 1, 1);
  var shade:Number = (penShade > 100) ? 200 - penShade : penShade; // range 0..100
  if (shade < 50) {
    penColorCache = Color.mixRGB(0, c, (10 + shade) / 60);
  } else {
    penColorCache = Color.mixRGB(c, 0xFFFFFF, (shade - 50) / 60);
  }
}
```

As far as I can tell, there's nothing *visibly* changing (assuming it's rendering as 0xRRGGBB), but you can see in my project that the actual value is changing - otherwise the color/shade would stay the same forever: `int(50 + 0.2) == 50`.